src/uniq_iter.rs

```diff
@@ -0,0 +1,15 @@
+struct UniqueIter {}
```

Comment:
How does this not constitute premature abstraction? :)

Also I think a change like this does require support from benchmarks showing that this does not reduce performance.

If it does not change performance, I am ok with a bit of overabstraction ;)